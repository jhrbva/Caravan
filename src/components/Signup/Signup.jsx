import React from 'react';
import { Field, Form, Formik } from 'formik';
import { withRouter } from 'react-router';
import axios from 'axios';

import '../../utilities/fonts.scss';
import Input from '../Input/Input';
import BigButton from '../BigButton/BigButton';
import {
	validateEmail,
	validatePhone,
	required,
	validatePassword,
} from '../../utilities/formValidation';

export const SignupForm = () => (
	<Form>
		<Field
			icon='person_pin'
			type='text'
			name='firstname'
			placeholder='First Name'
			validate={required}
			autoFocus={true}
			component={Input}
		/>
		<Field
			icon='person_pin'
			type='text'
			name='lastname'
			placeholder='Last Name'
			validate={required}
			autoFocus={false}
			component={Input}
		/>
		{/* TO DO: error message in case username already exists */}
		<Field
			icon='person_pin'
			type='text'
			name='username'
			placeholder='Username'
			validate={required}
			autoFocus={false}
			component={Input}
		/>
		<Field
			icon='call'
			type='text'
			name='phone'
			placeholder='Phone Number'
			validate={validatePhone}
			autoFocus={false}
			component={Input}
		/>
		<Field
			icon='email'
			type='text'
			name='email'
			placeholder='Email'
			validate={validateEmail}
			autoFocus={false}
			component={Input}
		/>
		<Field
			icon='lock'
			type='password'
			name='password'
			placeholder='Password'
			validate={validatePassword}
			autoFocus={false}
			component={Input}
		/>

		<BigButton value={'Sign up'} />
	</Form>
);

const Signup = props => {
	return (
		<div>
			<Formik
				initialValues={{
					firstname: '',
					lastname: '',
					username: '',
					phone: '',
					email: '',
					password: '',
				}}
				onSubmit={values => {
					const { history } = props;
					// same shape as initial values
					console.log(values);
					console.log(props);
					axios
						.post('/signup', {
							firstname: values.firstname,
							lastname: values.lastname,
							username: values.username,
							email: values.email,
							phonenumber: values.phone,
							password: values.password,
						})
						.then(function(response) {
							console.log(response);
							history.push('/trip');
						})
						.catch(function(error) {
							console.log(error);
						});
				}}
			>
				{() => <SignupForm />}
			</Formik>
			<p>Already have an account?</p>
			<a href='/'>Sign In</a>
		</div>
	);
};

export default withRouter(Signup);
