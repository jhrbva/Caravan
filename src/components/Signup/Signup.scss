@import "../../utilities/colorPalette.scss";

.signup-page-wrapper {
	margin: 15px;
	width: 20%;
	p {
		text-align: center;
		font-size: 16px;
	}
}
