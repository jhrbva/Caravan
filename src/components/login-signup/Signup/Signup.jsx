import React from "react";
import { Field, Form, Formik } from "formik";

import "../../../utilities/fonts.scss";
import Input from "../Input/Input";
import BigButton from "../BigButton/BigButton";
import {
	validateEmail,
	validatePhone,
	required,
} from "../../../utilities/formValidation";

const Signup = () => (
	<div>
		<Formik
			initialValues={{
				email: "",
				phone: "",
			}}
			onSubmit={values => {
				// same shape as initial values
				console.log(values);
			}}
		>
			{() => (
				<Form>
					<Field
						icon='person_pin'
						type='text'
						name='first_name'
						placeholder='First Name'
						validate={required}
						autofocus='true'
						component={Input}
					/>
					<Field
						icon='person_pin'
						type='text'
						name='last_name'
						placeholder='Last Name'
						validate={required}
						autofocus='true'
						component={Input}
					/>
					<Field
						icon='call'
						type='text'
						name='phone'
						placeholder='Phone Number'
						validate={validatePhone}
						autofocus='true'
						component={Input}
					/>
					<Field
						icon='email'
						type='text'
						name='email'
						placeholder='Email'
						validate={validateEmail}
						autofocus='true'
						component={Input}
					/>
					<Field
						icon='lock'
						type='text'
						name='password'
						placeholder='Password'
						validate={required}
						autofocus='true'
						component={Input}
					/>
					<BigButton value={"Signup"} />
				</Form>
			)}
		</Formik>
	</div>
);
export default Signup;
