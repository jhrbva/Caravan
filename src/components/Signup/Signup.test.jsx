import React from 'react';
import { shallow } from 'enzyme';
import { SignupForm, Signup } from './Signup';

describe('Signup', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<Signup />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('SignupForm', () => {
	it('should have five fields', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		expect(fieldWrapper.length).toBe(6);
	});
	it('should have an first name field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const firstNameField = fieldWrapper.find({ name: 'firstname' });
		expect(firstNameField.length).toBe(1);
	});
	it('should have an last name field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const lastNameField = fieldWrapper.find({ name: 'lastname' });
		expect(lastNameField.length).toBe(1);
	});
	it('should have an username field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'username' });
		expect(usernameField.length).toBe(1);
	});
	it('should have an phone field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const phoneField = fieldWrapper.find({ name: 'phone' });
		expect(phoneField.length).toBe(1);
	});
	it('should have an email field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const emailField = fieldWrapper.find({ name: 'email' });
		expect(emailField.length).toBe(1);
	});
	it('should have a password field', () => {
		const wrapper = shallow(<SignupForm />);
		const fieldWrapper = wrapper.find('Field');
		const passwordField = fieldWrapper.find({ name: 'password' });
		expect(passwordField.length).toBe(1);
	});
});
