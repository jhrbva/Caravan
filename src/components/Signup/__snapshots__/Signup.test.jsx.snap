// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match the snapshot 1`] = `
<div>
  <img
    alt="Caravan logo"
    className="caravan-logo"
    src="caravan-logo-blueOnWhite.png"
  />
  <Formik
    initialValues={
      Object {
        "email": "",
        "firstname": "",
        "lastname": "",
        "password": "",
        "phone": "",
        "username": "",
      }
    }
    onSubmit={[Function]}
  >
    <Component />
  </Formik>
  <p>
    Already have an account?
  </p>
  <a
    href="/"
  >
    Sign In
  </a>
</div>
`;
